Several test_source_cli failures with Python 3.14

We're seeing a number of test failures with Python 3.14, as in https://people.debian.org/~sanvila/build-logs/202512/pydantic-settings_2.12.0-1_amd64-20251216T152409.177Z.  I reproduced these locally with current `main` (cab6fc0f176e5cc095cf143c47b0f6d70a07cb88):

<details>
<summary>Test output</summary>

```
============================================================================= test session starts ==============================================================================
platform linux -- Python 3.14.2, pytest-9.0.2, pluggy-1.6.0
rootdir: /home/cjwatson/src/python/pydantic-settings
configfile: pyproject.toml
plugins: examples-0.0.18, mock-3.15.1, pretty-1.3.0
collected 163 items

tests/test_source_cli.py ..............F                                 [  9%]
tests/test_source_cli.py:546 test_cli_help_union_of_models - Asserti…
tests/test_source_cli.py F                                               [  9%]
tests/test_source_cli.py:593 test_cli_help_default_or_none_model
tests/test_source_cli.py ..................F                             [ 21%]
tests/test_source_cli.py:1441 test_cli_enums - AssertionError: asser…
tests/test_source_cli.py ..................................s......ss.................................................................F [ 88%]
tests/test_source_cli.py:2333 test_cli_suppress - AssertionError: as…
tests/test_source_cli.py F                                               [ 88%]
tests/test_source_cli.py:2390 test_cli_mutually_exclusive_group
tests/test_source_cli.py ....F                                           [ 92%]
tests/test_source_cli.py:2596 test_cli_kebab_case - AssertionError: …
tests/test_source_cli.py .............                                                                                                                                   [100%]

=================================================================================== FAILURES ===================================================================================
________________________________________________________________________ test_cli_help_union_of_models _________________________________________________________________________

capsys = <_pytest.capture.CaptureFixture object at 0x7f59a96b4350>, monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f59a96dd710>

    def test_cli_help_union_of_models(capsys, monkeypatch):
        class Cat(BaseModel):
            meow: str = 'meow'

        class Dog(BaseModel):
            bark: str = 'bark'

        class Bird(BaseModel):
            caww: str = 'caww'
            tweet: str

        class Tiger(Cat):
            roar: str = 'roar'

        class Car(BaseSettings, cli_parse_args=True):
            driver: Cat | Dog | Bird = Tiger(meow='purr')

        with monkeypatch.context() as m:
            m.setattr(sys, 'argv', ['example.py', '--help'])

            if PYTHON_3_14:
                if IS_WINDOWS:
                    usage = '                            [--driver.bark str] [--driver.caww str]\n                            [--driver.tweet str]'
                else:
                    usage = '                         [--driver.bark str] [--driver.caww str]\n                         [--driver.tweet str]'
            else:
                usage = '                  [--driver.bark str] [--driver.caww str] [--driver.tweet str]'
            with pytest.raises(SystemExit):
                Car()
>           assert (
                sanitize_cli_output(capsys.readouterr().out)
                == f"""usage: example.py [-h] [--driver [JSON]] [--driver.meow str]
    {usage}

    {ARGPARSE_OPTIONS_TEXT}:
      -h, --help          show this help message and exit

    driver options:
      --driver [JSON]     set driver from JSON string (default: {{}})
      --driver.meow str   (default: purr)
      --driver.bark str   (default: bark)
      --driver.caww str   (default: caww)
      --driver.tweet str  (ifdef: required)
    """
            )
E           AssertionError: assert 'usage: examp...: required)\n' == 'usage: examp...: required)\n'
E
E             Skipping 69 identical leading characters in diff, use -v to show
E             -                  [--driver.bark str] [--driver.caww str]
E             ? -------
E             +           [--driver.bark str] [--driver.caww str] [--driver.tweet str]
E             ?                                              +++++++++++++++++++++
E             -                          [--driver.tweet str]...
E
E             ...Full output truncated (10 lines hidden), use '-vv' to show

tests/test_source_cli.py:576: AssertionError
_____________________________________________________________________ test_cli_help_default_or_none_model ______________________________________________________________________

capsys = <_pytest.capture.CaptureFixture object at 0x7f59a96b5750>, monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f59a95c87d0>

    def test_cli_help_default_or_none_model(capsys, monkeypatch):
        class DeeperSubModel(BaseModel):
            flag: bool

        class DeepSubModel(BaseModel):
            flag: bool
            deeper: DeeperSubModel | None = None

        class SubModel(BaseModel):
            flag: bool
            deep: DeepSubModel = DeepSubModel(flag=True)

        class Settings(BaseSettings, cli_parse_args=True):
            flag: bool = True
            sub_model: SubModel = SubModel(flag=False)
            opt_model: DeepSubModel | None = Field(None, description='Group Doc')
            fact_model: SubModel = Field(default_factory=lambda: SubModel(flag=True))

        with monkeypatch.context() as m:
            m.setattr(sys, 'argv', ['example.py', '--help'])

            if PYTHON_3_14:
                if IS_WINDOWS:
                    text = """                            [--sub_model.flag bool] [--sub_model.deep [JSON]]
                                [--sub_model.deep.flag bool]
                                [--sub_model.deep.deeper [{JSON,null}]]
                                [--sub_model.deep.deeper.flag bool]
                                [--opt_model [{JSON,null}]]
                                [--opt_model.flag bool]
                                [--opt_model.deeper [{JSON,null}]]
                                [--opt_model.deeper.flag bool]
                                [--fact_model [JSON]] [--fact_model.flag bool]
                                [--fact_model.deep [JSON]]
                                [--fact_model.deep.flag bool]
                                [--fact_model.deep.deeper [{JSON,null}]]
                                [--fact_model.deep.deeper.flag bool]
    """
                else:
                    text = """                         [--sub_model.flag bool] [--sub_model.deep [JSON]]
                             [--sub_model.deep.flag bool]
                             [--sub_model.deep.deeper [{JSON,null}]]
                             [--sub_model.deep.deeper.flag bool]
                             [--opt_model [{JSON,null}]] [--opt_model.flag bool]
                             [--opt_model.deeper [{JSON,null}]]
                             [--opt_model.deeper.flag bool] [--fact_model [JSON]]
                             [--fact_model.flag bool] [--fact_model.deep [JSON]]
                             [--fact_model.deep.flag bool]
                             [--fact_model.deep.deeper [{JSON,null}]]
                             [--fact_model.deep.deeper.flag bool]
    """
            else:
                text = """                  [--sub_model.flag bool] [--sub_model.deep [JSON]]
                      [--sub_model.deep.flag bool]
                      [--sub_model.deep.deeper [{JSON,null}]]
                      [--sub_model.deep.deeper.flag bool]
                      [--opt_model [{JSON,null}]] [--opt_model.flag bool]
                      [--opt_model.deeper [{JSON,null}]]
                      [--opt_model.deeper.flag bool] [--fact_model [JSON]]
                      [--fact_model.flag bool] [--fact_model.deep [JSON]]
                      [--fact_model.deep.flag bool]
                      [--fact_model.deep.deeper [{JSON,null}]]
                      [--fact_model.deep.deeper.flag bool]
    """
            with pytest.raises(SystemExit):
                Settings()
>           assert (
                sanitize_cli_output(capsys.readouterr().out)
                == f"""usage: example.py [-h] [--flag bool] [--sub_model [JSON]]
    {text}
    {ARGPARSE_OPTIONS_TEXT}:
      -h, --help            show this help message and exit
      --flag bool           (default: True)

    sub_model options:
      --sub_model [JSON]    set sub_model from JSON string (default: {{}})
      --sub_model.flag bool
                            (default: False)

    sub_model.deep options:
      --sub_model.deep [JSON]
                            set sub_model.deep from JSON string (default: {{}})
      --sub_model.deep.flag bool
                            (default: True)

    sub_model.deep.deeper options:
      default: null (undefined)

      --sub_model.deep.deeper [{{JSON,null}}]
                            set sub_model.deep.deeper from JSON string (default:
                            {{}})
      --sub_model.deep.deeper.flag bool
                            (ifdef: required)

    opt_model options:
      default: null (undefined)
      Group Doc

      --opt_model [{{JSON,null}}]
                            set opt_model from JSON string (default: {{}})
      --opt_model.flag bool
                            (ifdef: required)

    opt_model.deeper options:
      default: null (undefined)

      --opt_model.deeper [{{JSON,null}}]
                            set opt_model.deeper from JSON string (default: {{}})
      --opt_model.deeper.flag bool
                            (ifdef: required)

    fact_model options:
      --fact_model [JSON]   set fact_model from JSON string (default: {{}})
      --fact_model.flag bool
                            (default factory: <lambda>)

    fact_model.deep options:
      --fact_model.deep [JSON]
                            set fact_model.deep from JSON string (default: {{}})
      --fact_model.deep.flag bool
                            (default factory: <lambda>)

    fact_model.deep.deeper options:
      --fact_model.deep.deeper [{{JSON,null}}]
                            set fact_model.deep.deeper from JSON string (default:
                            {{}})
      --fact_model.deep.deeper.flag bool
                            (default factory: <lambda>)
    """
            )
E           AssertionError: assert 'usage: examp...: <lambda>)\n' == 'usage: examp...: <lambda>)\n'
E
E             Skipping 66 identical leading characters in diff, use -v to show
E             -                  [--sub_model.flag bool] [--sub_model.deep [JSON]]
E             ? -------
E             +           [--sub_model.flag bool] [--sub_model.deep [JSON]]
E             -                          [--sub_model.deep.flag bool]
E             ? -------...
E
E             ...Full output truncated (87 lines hidden), use '-vv' to show

tests/test_source_cli.py:659: AssertionError
________________________________________________________________________________ test_cli_enums ________________________________________________________________________________

capsys = <_pytest.capture.CaptureFixture object at 0x7f59a96523f0>, monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f59a9169350>

    def test_cli_enums(capsys, monkeypatch):
        class Pet(IntEnum):
            dog = 0
            cat = 1
            bird = 2
            crow = 2

        class Cfg(BaseSettings):
            pet: Pet = Pet.dog
            union_pet: Pet | int = 43

        cfg = CliApp.run(Cfg, cli_args=['--pet', 'cat', '--union_pet', 'dog'])
        assert cfg.model_dump() == {'pet': Pet.cat, 'union_pet': Pet.dog}

        cfg = CliApp.run(Cfg, cli_args=['--pet', 'crow', '--union_pet', 'dog'])
        assert cfg.model_dump() == {'pet': Pet.crow, 'union_pet': Pet.dog}

        with pytest.raises(ValidationError) as exc_info:
            CliApp.run(Cfg, cli_args=['--pet', 'rock'])
        assert exc_info.value.errors(include_url=False) == [
            {
                'type': 'enum',
                'loc': ('pet',),
                'msg': 'Input should be 0, 1, 2 or 2',
                'input': 'rock',
                'ctx': {'expected': '0, 1, 2 or 2'},
            }
        ]

        with monkeypatch.context() as m:
            m.setattr(sys, 'argv', ['example.py', '--help'])

            if PYTHON_3_14:
                if IS_WINDOWS:
                    text = '                            [--union_pet {{dog,cat,bird,crow},int}]'
                else:
                    text = '                         [--union_pet {{dog,cat,bird,crow},int}]'
            else:
                text = '                  [--union_pet {{dog,cat,bird,crow},int}]'
            with pytest.raises(SystemExit):
                CliApp.run(Cfg)
>           assert (
                sanitize_cli_output(capsys.readouterr().out)
                == f"""usage: example.py [-h] [--pet {{dog,cat,bird,crow}}]
    {text}

    {ARGPARSE_OPTIONS_TEXT}:
      -h, --help            show this help message and exit
      --pet {{dog,cat,bird,crow}}
                            (default: dog)
      --union_pet {{{{dog,cat,bird,crow}},int}}
                            (default: 43)
    """
            )
E           AssertionError: assert 'usage: examp...efault: 43)\n' == 'usage: examp...efault: 43)\n'
E
E             Skipping 59 identical leading characters in diff, use -v to show
E             -                  [--union_pet {{dog,cat,bird,crow},int}]
E             ? -------
E             +           [--union_pet {{dog,cat,bird,crow},int}]
E
E               options:...
E
E             ...Full output truncated (5 lines hidden), use '-vv' to show

tests/test_source_cli.py:1483: AssertionError
______________________________________________________________________________ test_cli_suppress _______________________________________________________________________________

capsys = <_pytest.capture.CaptureFixture object at 0x7f59a9163800>, monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f59a97224a0>

    def test_cli_suppress(capsys, monkeypatch):
        class DeepHiddenSubModel(BaseModel):
            deep_hidden_a: int
            deep_hidden_b: int

        class HiddenSubModel(BaseModel):
            hidden_a: int
            hidden_b: int
            deep_hidden_obj: DeepHiddenSubModel = Field(description='deep_hidden_obj description')

        class SubModel(BaseModel):
            visible_a: int
            visible_b: int
            deep_hidden_obj: CliSuppress[DeepHiddenSubModel] = Field(description='deep_hidden_obj description')

        class Settings(BaseSettings, cli_parse_args=True):
            field_a: CliSuppress[int] = 0
            field_b: str = Field(default='hi', description=CLI_SUPPRESS)
            hidden_obj: CliSuppress[HiddenSubModel] = Field(description='hidden_obj description')
            visible_obj: SubModel

        with monkeypatch.context() as m:
            m.setattr(sys, 'argv', ['example.py', '--help'])

            with pytest.raises(SystemExit):
                CliApp.run(Settings)

            if PYTHON_3_14:
                if IS_WINDOWS:
                    text = """usage: example.py [-h] [--visible_obj [JSON]]
                                [--visible_obj.visible_a int]
                                [--visible_obj.visible_b int]"""
                else:
                    text = """usage: example.py [-h] [--visible_obj [JSON]]
                             [--visible_obj.visible_a int]
                             [--visible_obj.visible_b int]"""

            else:
                text = """usage: example.py [-h] [--visible_obj [JSON]] [--visible_obj.visible_a int]
                      [--visible_obj.visible_b int]"""
>           assert (
                sanitize_cli_output(capsys.readouterr().out)
                == f"""{text}

    {ARGPARSE_OPTIONS_TEXT}:
      -h, --help            show this help message and exit

    visible_obj options:
      --visible_obj [JSON]  set visible_obj from JSON string (default: {{}})
      --visible_obj.visible_a int
                            (required)
      --visible_obj.visible_b int
                            (required)
    """
            )
E           AssertionError: assert 'usage: examp... (required)\n' == 'usage: examp... (required)\n'
E
E             Skipping 35 identical leading characters in diff, use -v to show
E             + bj [JSON]] [--visible_obj.visible_a int]
E             - bj [JSON]]
E             -                          [--visible_obj.visible_a int]
E             -                          [--visible_obj.visible_b int]
E             ? -------...
E
E             ...Full output truncated (11 lines hidden), use '-vv' to show

tests/test_source_cli.py:2374: AssertionError
______________________________________________________________________ test_cli_mutually_exclusive_group _______________________________________________________________________

capsys = <_pytest.capture.CaptureFixture object at 0x7f59a9079440>, monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f59a969b690>

    def test_cli_mutually_exclusive_group(capsys, monkeypatch):
        class Circle(CliMutuallyExclusiveGroup):
            radius: float | None = 21
            diameter: float | None = 22
            perimeter: float | None = 23

        class Settings(BaseModel):
            circle_optional: Circle = Circle(radius=None, diameter=None, perimeter=24)
            circle_required: Circle

        CliApp.run(Settings, cli_args=['--circle-required.radius=1', '--circle-optional.radius=1']).model_dump() == {
            'circle_optional': {'radius': 1, 'diameter': 22, 'perimeter': 24},
            'circle_required': {'radius': 1, 'diameter': 22, 'perimeter': 23},
        }

        with pytest.raises(SystemExit):
            CliApp.run(Settings, cli_args=['--circle-required.radius=1', '--circle-required.diameter=2'])
        assert (
            'error: argument --circle-required.diameter: not allowed with argument --circle-required.radius'
            in capsys.readouterr().err
        )

        with pytest.raises(SystemExit):
            CliApp.run(
                Settings,
                cli_args=['--circle-required.radius=1', '--circle-optional.radius=1', '--circle-optional.diameter=2'],
            )
        assert (
            'error: argument --circle-optional.diameter: not allowed with argument --circle-optional.radius'
            in capsys.readouterr().err
        )

        with monkeypatch.context() as m:
            m.setattr(sys, 'argv', ['example.py', '--help'])
            with pytest.raises(SystemExit):
                CliApp.run(Settings)
            if PYTHON_3_14:
                if IS_WINDOWS:
                    usage = """usage: example.py [-h] [--circle-optional.radius float |
                                --circle-optional.diameter float |
                                --circle-optional.perimeter float]
                                (--circle-required.radius float |
                                --circle-required.diameter float |
                                --circle-required.perimeter float)"""
                else:
                    usage = """usage: example.py [-h] [--circle-optional.radius float |
                             --circle-optional.diameter float |
                             --circle-optional.perimeter float]
                             (--circle-required.radius float |
                             --circle-required.diameter float |
                             --circle-required.perimeter float)"""
            elif sys.version_info >= (3, 13):
                usage = """usage: example.py [-h] [--circle-optional.radius float |
                      --circle-optional.diameter float |
                      --circle-optional.perimeter float]
                      (--circle-required.radius float |
                      --circle-required.diameter float |
                      --circle-required.perimeter float)"""
            else:
                usage = """usage: example.py [-h]
                      [--circle-optional.radius float | --circle-optional.diameter float | --circle-optional.perimeter float]
                      (--circle-required.radius float | --circle-required.diameter float | --circle-required.perimeter float)"""
>           assert (
                sanitize_cli_output(capsys.readouterr().out)
                == f"""{usage}

    {ARGPARSE_OPTIONS_TEXT}:
      -h, --help            show this help message and exit

    circle-optional options (mutually exclusive):
      --circle-optional.radius float
                            (default: None)
      --circle-optional.diameter float
                            (default: None)
      --circle-optional.perimeter float
                            (default: 24.0)

    circle-required options (mutually exclusive):
      --circle-required.radius float
                            (default: 21)
      --circle-required.diameter float
                            (default: 22)
      --circle-required.perimeter float
                            (default: 23)
    """
            )
E           AssertionError: assert 'usage: examp...efault: 23)\n' == 'usage: examp...efault: 23)\n'
E
E             Skipping 65 identical leading characters in diff, use -v to show
E             -                  --circle-optional.diameter float |
E             ? -------
E             +           --circle-optional.diameter float |
E             -                          --circle-optional.perimeter float]
E             ? -------...
E
E             ...Full output truncated (29 lines hidden), use '-vv' to show

tests/test_source_cli.py:2453: AssertionError
_____________________________________________________________________________ test_cli_kebab_case ______________________________________________________________________________

capsys = <_pytest.capture.CaptureFixture object at 0x7f59a9549270>, monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f59a97bacf0>

    def test_cli_kebab_case(capsys, monkeypatch):
        class DeepSubModel(BaseModel):
            deep_pos_arg: CliPositionalArg[str]
            deep_arg: str

        class SubModel(BaseModel):
            sub_subcmd: CliSubCommand[DeepSubModel]
            sub_other_subcmd: CliSubCommand[DeepSubModel]
            sub_arg: str

        class Root(BaseModel):
            root_subcmd: CliSubCommand[SubModel]
            other_subcmd: CliSubCommand[SubModel]
            root_arg: str

        root = CliApp.run(
            Root,
            cli_args=[
                '--root-arg=hi',
                'root-subcmd',
                '--sub-arg=hello',
                'sub-subcmd',
                'hey',
                '--deep-arg=bye',
            ],
        )
        assert root.model_dump() == {
            'root_arg': 'hi',
            'other_subcmd': None,
            'root_subcmd': {
                'sub_arg': 'hello',
                'sub_subcmd': {'deep_pos_arg': 'hey', 'deep_arg': 'bye'},
                'sub_other_subcmd': None,
            },
        }

        serialized_cli_args = CliApp.serialize(root)
        assert serialized_cli_args == [
            '--root-arg',
            'hi',
            'root-subcmd',
            '--sub-arg',
            'hello',
            'sub-subcmd',
            '--deep-arg',
            'bye',
            'hey',
        ]

        with monkeypatch.context() as m:
            m.setattr(sys, 'argv', ['example.py', '--help'])
            with pytest.raises(SystemExit):
                CliApp.run(Root)
            assert (
                sanitize_cli_output(capsys.readouterr().out)
                == f"""usage: example.py [-h] --root-arg str {{root-subcmd,other-subcmd}} ...

    {ARGPARSE_OPTIONS_TEXT}:
      -h, --help            show this help message and exit
      --root-arg str        (required)

    subcommands:
      {{root-subcmd,other-subcmd}}
        root-subcmd
        other-subcmd
    """
            )

            if PYTHON_3_14:
                if IS_WINDOWS:
                    usage = """usage: example.py root-subcmd [-h] --sub-arg str
                                            {sub-subcmd,sub-other-subcmd} ..."""
                else:
                    usage = """usage: example.py root-subcmd [-h] --sub-arg str
                                         {sub-subcmd,sub-other-subcmd} ..."""
            else:
                usage = """usage: example.py root-subcmd [-h] --sub-arg str
                                  {sub-subcmd,sub-other-subcmd} ..."""
            m.setattr(sys, 'argv', ['example.py', 'root-subcmd', '--help'])
            with pytest.raises(SystemExit):
                CliApp.run(Root)
>           assert (
                sanitize_cli_output(capsys.readouterr().out)
                == f"""{usage}

    {ARGPARSE_OPTIONS_TEXT}:
      -h, --help            show this help message and exit
      --sub-arg str         (required)

    subcommands:
      {{sub-subcmd,sub-other-subcmd}}
        sub-subcmd
        sub-other-subcmd
    """
            )
E           AssertionError: assert 'usage: examp...ther-subcmd\n' == 'usage: examp...ther-subcmd\n'
E
E             Skipping 69 identical leading characters in diff, use -v to show
E             -                  {sub-subcmd,sub-other-subcmd} ...
E             ? -------
E             +           {sub-subcmd,sub-other-subcmd} ...
E
E               options:...
E
E             ...Full output truncated (7 lines hidden), use '-vv' to show

tests/test_source_cli.py:2678: AssertionError
                                                   Summary of Failures
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━┓
┃  File                      ┃  Function                             ┃  Function Line  ┃  Error Line  ┃  Error           ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━┩
│  tests/test_source_cli.py  │  test_cli_help_union_of_models        │  547            │  576         │  AssertionError  │
│  tests/test_source_cli.py  │  test_cli_help_default_or_none_model  │  594            │  659         │  AssertionError  │
│  tests/test_source_cli.py  │  test_cli_enums                       │  1442           │  1483        │  AssertionError  │
│  tests/test_source_cli.py  │  test_cli_suppress                    │  2334           │  2374        │  AssertionError  │
│  tests/test_source_cli.py  │  test_cli_mutually_exclusive_group    │  2391           │  2453        │  AssertionError  │
│  tests/test_source_cli.py  │  test_cli_kebab_case                  │  2597           │  2678        │  AssertionError  │
└────────────────────────────┴───────────────────────────────────────┴─────────────────┴──────────────┴──────────────────┘
Results (0.54s):
         6 failed
       154 passed
         3 skipped
```

</details>

I guess `argparse` has changed slightly, but I'm not sure why it apparently worked in #704 but fails now.

--- a/tests/test_source_cli.py
+++ b/tests/test_source_cli.py
@@ -55,14 +55,6 @@ from pydantic_settings.sources import (
 )
 
 ARGPARSE_OPTIONS_TEXT = 'options' if sys.version_info >= (3, 10) else 'optional arguments'
-PYTHON_3_14 = sys.version_info >= (3, 14)
-IS_WINDOWS = sys.platform.startswith('win')
-
-
-def sanitize_cli_output(output: str) -> str:
-    output = re.sub(r'\bpython3? ', '', output)
-    output = output.replace('python.exe -m pytest', 'example.py')  # For Windows compatibility
-    return output
 
 
 @pytest.fixture(autouse=True)
@@ -242,7 +234,7 @@ def test_cli_alias_subcommand_and_positional_args(capsys, monkeypatch):
     class SubCmd(BaseModel):
         pos_arg: CliPositionalArg[str] = Field(validation_alias='pos-arg')
 
-    class Cfg(BaseSettings):
+    class Cfg(BaseSettings, cli_prog_name='example.py'):
         sub_cmd: CliSubCommand[SubCmd] = Field(validation_alias='sub-cmd')
 
     cfg = Cfg(**{'sub-cmd': {'pos-arg': 'howdy'}})
@@ -257,7 +249,7 @@ def test_cli_alias_subcommand_and_positional_args(capsys, monkeypatch):
         with pytest.raises(SystemExit):
             CliApp.run(Cfg)
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h] {{sub-cmd}} ...
 
 {ARGPARSE_OPTIONS_TEXT}:
@@ -273,7 +265,7 @@ subcommands:
         with pytest.raises(SystemExit):
             CliApp.run(Cfg)
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py sub-cmd [-h] POS-ARG
 
 positional arguments:
@@ -463,7 +455,7 @@ def test_cli_case_insensitive_arg():
 
 
 def test_cli_help_differentiation(capsys, monkeypatch):
-    class Cfg(BaseSettings):
+    class Cfg(BaseSettings, cli_prog_name='example.py'):
         foo: str
         bar: int = 123
         boo: int = Field(default_factory=lambda: 456)
@@ -475,7 +467,7 @@ def test_cli_help_differentiation(capsys, monkeypatch):
             CliApp.run(Cfg)
 
         assert (
-            re.sub(r'0x\w+', '0xffffffff', sanitize_cli_output(capsys.readouterr().out), flags=re.MULTILINE)
+            re.sub(r'0x\w+', '0xffffffff', capsys.readouterr().out, flags=re.MULTILINE)
             == f"""usage: example.py [-h] [--foo str] [--bar int] [--boo int]
 
 {ARGPARSE_OPTIONS_TEXT}:
@@ -488,10 +480,10 @@ def test_cli_help_differentiation(capsys, monkeypatch):
 
 
 def test_cli_help_string_format(capsys, monkeypatch):
-    class Cfg(BaseSettings, cli_parse_args=True):
+    class Cfg(BaseSettings, cli_parse_args=True, cli_prog_name='example.py'):
         date_str: str = '%Y-%m-%d'
 
-    class MultilineDoc(BaseSettings, cli_parse_args=True):
+    class MultilineDoc(BaseSettings, cli_parse_args=True, cli_prog_name='example.py'):
         """
         My
         Multiline
@@ -505,7 +497,7 @@ def test_cli_help_string_format(capsys, monkeypatch):
             Cfg()
 
         assert (
-            re.sub(r'0x\w+', '0xffffffff', sanitize_cli_output(capsys.readouterr().out), flags=re.MULTILINE)
+            re.sub(r'0x\w+', '0xffffffff', capsys.readouterr().out, flags=re.MULTILINE)
             == f"""usage: example.py [-h] [--date_str str]
 
 {ARGPARSE_OPTIONS_TEXT}:
@@ -517,7 +509,7 @@ def test_cli_help_string_format(capsys, monkeypatch):
         with pytest.raises(SystemExit):
             MultilineDoc()
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h]
 
 My
@@ -533,7 +525,7 @@ Doc
             cli_settings_source = CliSettingsSource(MultilineDoc, formatter_class=argparse.HelpFormatter)
             MultilineDoc(_cli_settings_source=cli_settings_source(args=True))
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h]
 
 My Multiline Doc
@@ -558,25 +550,18 @@ def test_cli_help_union_of_models(capsys, monkeypatch):
     class Tiger(Cat):
         roar: str = 'roar'
 
-    class Car(BaseSettings, cli_parse_args=True):
+    class Car(BaseSettings, cli_parse_args=True, cli_prog_name='example.py'):
         driver: Cat | Dog | Bird = Tiger(meow='purr')
 
     with monkeypatch.context() as m:
         m.setattr(sys, 'argv', ['example.py', '--help'])
 
-        if PYTHON_3_14:
-            if IS_WINDOWS:
-                usage = '                            [--driver.bark str] [--driver.caww str]\n                            [--driver.tweet str]'
-            else:
-                usage = '                         [--driver.bark str] [--driver.caww str]\n                         [--driver.tweet str]'
-        else:
-            usage = '                  [--driver.bark str] [--driver.caww str] [--driver.tweet str]'
         with pytest.raises(SystemExit):
             Car()
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h] [--driver [JSON]] [--driver.meow str]
-{usage}
+                  [--driver.bark str] [--driver.caww str] [--driver.tweet str]
 
 {ARGPARSE_OPTIONS_TEXT}:
   -h, --help          show this help message and exit
@@ -603,7 +588,7 @@ def test_cli_help_default_or_none_model(capsys, monkeypatch):
         flag: bool
         deep: DeepSubModel = DeepSubModel(flag=True)
 
-    class Settings(BaseSettings, cli_parse_args=True):
+    class Settings(BaseSettings, cli_parse_args=True, cli_prog_name='example.py'):
         flag: bool = True
         sub_model: SubModel = SubModel(flag=False)
         opt_model: DeepSubModel | None = Field(None, description='Group Doc')
@@ -612,54 +597,23 @@ def test_cli_help_default_or_none_model(capsys, monkeypatch):
     with monkeypatch.context() as m:
         m.setattr(sys, 'argv', ['example.py', '--help'])
 
-        if PYTHON_3_14:
-            if IS_WINDOWS:
-                text = """                            [--sub_model.flag bool] [--sub_model.deep [JSON]]
-                            [--sub_model.deep.flag bool]
-                            [--sub_model.deep.deeper [{JSON,null}]]
-                            [--sub_model.deep.deeper.flag bool]
-                            [--opt_model [{JSON,null}]]
-                            [--opt_model.flag bool]
-                            [--opt_model.deeper [{JSON,null}]]
-                            [--opt_model.deeper.flag bool]
-                            [--fact_model [JSON]] [--fact_model.flag bool]
-                            [--fact_model.deep [JSON]]
-                            [--fact_model.deep.flag bool]
-                            [--fact_model.deep.deeper [{JSON,null}]]
-                            [--fact_model.deep.deeper.flag bool]
-"""
-            else:
-                text = """                         [--sub_model.flag bool] [--sub_model.deep [JSON]]
-                         [--sub_model.deep.flag bool]
-                         [--sub_model.deep.deeper [{JSON,null}]]
-                         [--sub_model.deep.deeper.flag bool]
-                         [--opt_model [{JSON,null}]] [--opt_model.flag bool]
-                         [--opt_model.deeper [{JSON,null}]]
-                         [--opt_model.deeper.flag bool] [--fact_model [JSON]]
-                         [--fact_model.flag bool] [--fact_model.deep [JSON]]
-                         [--fact_model.deep.flag bool]
-                         [--fact_model.deep.deeper [{JSON,null}]]
-                         [--fact_model.deep.deeper.flag bool]
-"""
-        else:
-            text = """                  [--sub_model.flag bool] [--sub_model.deep [JSON]]
-                  [--sub_model.deep.flag bool]
-                  [--sub_model.deep.deeper [{JSON,null}]]
-                  [--sub_model.deep.deeper.flag bool]
-                  [--opt_model [{JSON,null}]] [--opt_model.flag bool]
-                  [--opt_model.deeper [{JSON,null}]]
-                  [--opt_model.deeper.flag bool] [--fact_model [JSON]]
-                  [--fact_model.flag bool] [--fact_model.deep [JSON]]
-                  [--fact_model.deep.flag bool]
-                  [--fact_model.deep.deeper [{JSON,null}]]
-                  [--fact_model.deep.deeper.flag bool]
-"""
         with pytest.raises(SystemExit):
             Settings()
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h] [--flag bool] [--sub_model [JSON]]
-{text}
+                  [--sub_model.flag bool] [--sub_model.deep [JSON]]
+                  [--sub_model.deep.flag bool]
+                  [--sub_model.deep.deeper [{{JSON,null}}]]
+                  [--sub_model.deep.deeper.flag bool]
+                  [--opt_model [{{JSON,null}}]] [--opt_model.flag bool]
+                  [--opt_model.deeper [{{JSON,null}}]]
+                  [--opt_model.deeper.flag bool] [--fact_model [JSON]]
+                  [--fact_model.flag bool] [--fact_model.deep [JSON]]
+                  [--fact_model.deep.flag bool]
+                  [--fact_model.deep.deeper [{{JSON,null}}]]
+                  [--fact_model.deep.deeper.flag bool]
+
 {ARGPARSE_OPTIONS_TEXT}:
   -h, --help            show this help message and exit
   --flag bool           (default: True)
@@ -1111,7 +1065,7 @@ def test_cli_subcommand_union(capsys, monkeypatch):
 
         g: str
 
-    class Root1(BaseSettings):
+    class Root1(BaseSettings, cli_prog_name='example.py'):
         """Root Help"""
 
         subcommand: CliSubCommand[AlphaCmd | BetaCmd | GammaCmd] = Field(description='Field Help')
@@ -1132,7 +1086,7 @@ def test_cli_subcommand_union(capsys, monkeypatch):
         with pytest.raises(SystemExit):
             CliApp.run(Root1)
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h] {{AlphaCmd,BetaCmd,GammaCmd}} ...
 
 Root Help
@@ -1153,7 +1107,7 @@ subcommands:
         with pytest.raises(SystemExit):
             Root1(_cli_parse_args=True, _cli_use_class_docs_for_groups=True)
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h] {{AlphaCmd,BetaCmd,GammaCmd}} ...
 
 Root Help
@@ -1171,7 +1125,7 @@ subcommands:
 """
         )
 
-    class Root2(BaseSettings):
+    class Root2(BaseSettings, cli_prog_name='example.py'):
         """Root Help"""
 
         subcommand: CliSubCommand[AlphaCmd | GammaCmd] = Field(description='Field Help')
@@ -1193,7 +1147,7 @@ subcommands:
         with pytest.raises(SystemExit):
             CliApp.run(Root2, cli_args=True)
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h] {{AlphaCmd,GammaCmd,beta}} ...
 
 Root Help
@@ -1214,7 +1168,7 @@ subcommands:
         with pytest.raises(SystemExit):
             Root2(_cli_parse_args=True, _cli_use_class_docs_for_groups=True)
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h] {{AlphaCmd,GammaCmd,beta}} ...
 
 Root Help
@@ -1232,7 +1186,7 @@ subcommands:
 """
         )
 
-    class Root3(BaseSettings):
+    class Root3(BaseSettings, cli_prog_name='example.py'):
         """Root Help"""
 
         beta: CliSubCommand[BetaCmd] = Field(description='Field Beta Help')
@@ -1254,7 +1208,7 @@ subcommands:
         with pytest.raises(SystemExit):
             CliApp.run(Root3)
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h] {{beta,AlphaCmd,GammaCmd}} ...
 
 Root Help
@@ -1273,7 +1227,7 @@ subcommands:
         with pytest.raises(SystemExit):
             Root3(_cli_parse_args=True, _cli_use_class_docs_for_groups=True)
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h] {{beta,AlphaCmd,GammaCmd}} ...
 
 Root Help
@@ -1446,7 +1400,7 @@ def test_cli_enums(capsys, monkeypatch):
         bird = 2
         crow = 2
 
-    class Cfg(BaseSettings):
+    class Cfg(BaseSettings, cli_prog_name='example.py'):
         pet: Pet = Pet.dog
         union_pet: Pet | int = 43
 
@@ -1471,19 +1425,12 @@ def test_cli_enums(capsys, monkeypatch):
     with monkeypatch.context() as m:
         m.setattr(sys, 'argv', ['example.py', '--help'])
 
-        if PYTHON_3_14:
-            if IS_WINDOWS:
-                text = '                            [--union_pet {{dog,cat,bird,crow},int}]'
-            else:
-                text = '                         [--union_pet {{dog,cat,bird,crow},int}]'
-        else:
-            text = '                  [--union_pet {{dog,cat,bird,crow},int}]'
         with pytest.raises(SystemExit):
             CliApp.run(Cfg)
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h] [--pet {{dog,cat,bird,crow}}]
-{text}
+                  [--union_pet {{{{dog,cat,bird,crow}},int}}]
 
 {ARGPARSE_OPTIONS_TEXT}:
   -h, --help            show this help message and exit
@@ -1704,7 +1651,7 @@ def test_cli_avoid_json(capsys, monkeypatch):
     class SubModel(BaseModel):
         v1: int
 
-    class Settings(BaseSettings):
+    class Settings(BaseSettings, cli_prog_name='example.py'):
         sub_model: SubModel
 
         model_config = SettingsConfigDict(cli_parse_args=True)
@@ -1716,7 +1663,7 @@ def test_cli_avoid_json(capsys, monkeypatch):
             Settings(_cli_avoid_json=False)
 
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h] [--sub_model [JSON]] [--sub_model.v1 int]
 
 {ARGPARSE_OPTIONS_TEXT}:
@@ -1732,7 +1679,7 @@ sub_model options:
             Settings(_cli_avoid_json=True)
 
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h] [--sub_model.v1 int]
 
 {ARGPARSE_OPTIONS_TEXT}:
@@ -1748,7 +1695,7 @@ def test_cli_remove_empty_groups(capsys, monkeypatch):
     class SubModel(BaseModel):
         pass
 
-    class Settings(BaseSettings):
+    class Settings(BaseSettings, cli_prog_name='example.py'):
         sub_model: SubModel
 
         model_config = SettingsConfigDict(cli_parse_args=True)
@@ -1760,7 +1707,7 @@ def test_cli_remove_empty_groups(capsys, monkeypatch):
             Settings(_cli_avoid_json=False)
 
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h] [--sub_model [JSON]]
 
 {ARGPARSE_OPTIONS_TEXT}:
@@ -1775,7 +1722,7 @@ sub_model options:
             Settings(_cli_avoid_json=True)
 
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h]
 
 {ARGPARSE_OPTIONS_TEXT}:
@@ -1785,7 +1732,7 @@ sub_model options:
 
 
 def test_cli_hide_none_type(capsys, monkeypatch):
-    class Settings(BaseSettings):
+    class Settings(BaseSettings, cli_prog_name='example.py'):
         v0: str | None
 
         model_config = SettingsConfigDict(cli_parse_args=True)
@@ -1797,7 +1744,7 @@ def test_cli_hide_none_type(capsys, monkeypatch):
             Settings(_cli_hide_none_type=False)
 
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h] [--v0 {{str,null}}]
 
 {ARGPARSE_OPTIONS_TEXT}:
@@ -1810,7 +1757,7 @@ def test_cli_hide_none_type(capsys, monkeypatch):
             Settings(_cli_hide_none_type=True)
 
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h] [--v0 str]
 
 {ARGPARSE_OPTIONS_TEXT}:
@@ -1826,7 +1773,7 @@ def test_cli_use_class_docs_for_groups(capsys, monkeypatch):
 
         v1: int
 
-    class Settings(BaseSettings):
+    class Settings(BaseSettings, cli_prog_name='example.py'):
         """My application help text."""
 
         sub_model: SubModel = Field(description='The help text from the field description')
@@ -1840,7 +1787,7 @@ def test_cli_use_class_docs_for_groups(capsys, monkeypatch):
             Settings(_cli_use_class_docs_for_groups=False)
 
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h] [--sub_model [JSON]] [--sub_model.v1 int]
 
 My application help text.
@@ -1860,7 +1807,7 @@ sub_model options:
             Settings(_cli_use_class_docs_for_groups=True)
 
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h] [--sub_model [JSON]] [--sub_model.v1 int]
 
 My application help text.
@@ -1905,7 +1852,7 @@ def test_cli_enforce_required(env):
 
 
 def test_cli_exit_on_error(capsys, monkeypatch):
-    class Settings(BaseSettings, cli_parse_args=True): ...
+    class Settings(BaseSettings, cli_parse_args=True, cli_prog_name='example.py'): ...
 
     with monkeypatch.context() as m:
         m.setattr(sys, 'argv', ['example.py', '--bad-arg'])
@@ -1913,7 +1860,7 @@ def test_cli_exit_on_error(capsys, monkeypatch):
         with pytest.raises(SystemExit):
             Settings()
         assert (
-            sanitize_cli_output(capsys.readouterr().err)
+            capsys.readouterr().err
             == """usage: example.py [-h]
 example.py: error: unrecognized arguments: --bad-arg
 """
@@ -2346,7 +2293,7 @@ def test_cli_suppress(capsys, monkeypatch):
         visible_b: int
         deep_hidden_obj: CliSuppress[DeepHiddenSubModel] = Field(description='deep_hidden_obj description')
 
-    class Settings(BaseSettings, cli_parse_args=True):
+    class Settings(BaseSettings, cli_parse_args=True, cli_prog_name='example.py'):
         field_a: CliSuppress[int] = 0
         field_b: str = Field(default='hi', description=CLI_SUPPRESS)
         hidden_obj: CliSuppress[HiddenSubModel] = Field(description='hidden_obj description')
@@ -2358,22 +2305,10 @@ def test_cli_suppress(capsys, monkeypatch):
         with pytest.raises(SystemExit):
             CliApp.run(Settings)
 
-        if PYTHON_3_14:
-            if IS_WINDOWS:
-                text = """usage: example.py [-h] [--visible_obj [JSON]]
-                            [--visible_obj.visible_a int]
-                            [--visible_obj.visible_b int]"""
-            else:
-                text = """usage: example.py [-h] [--visible_obj [JSON]]
-                         [--visible_obj.visible_a int]
-                         [--visible_obj.visible_b int]"""
-
-        else:
-            text = """usage: example.py [-h] [--visible_obj [JSON]] [--visible_obj.visible_a int]
-                  [--visible_obj.visible_b int]"""
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
-            == f"""{text}
+            capsys.readouterr().out
+            == f"""usage: example.py [-h] [--visible_obj [JSON]] [--visible_obj.visible_a int]
+                  [--visible_obj.visible_b int]
 
 {ARGPARSE_OPTIONS_TEXT}:
   -h, --help            show this help message and exit
@@ -2394,7 +2329,7 @@ def test_cli_mutually_exclusive_group(capsys, monkeypatch):
         diameter: float | None = 22
         perimeter: float | None = 23
 
-    class Settings(BaseModel):
+    class Settings(BaseModel, cli_prog_name='example.py'):
         circle_optional: Circle = Circle(radius=None, diameter=None, perimeter=24)
         circle_required: Circle
 
@@ -2424,34 +2359,20 @@ def test_cli_mutually_exclusive_group(capsys, monkeypatch):
         m.setattr(sys, 'argv', ['example.py', '--help'])
         with pytest.raises(SystemExit):
             CliApp.run(Settings)
-        if PYTHON_3_14:
-            if IS_WINDOWS:
-                usage = """usage: example.py [-h] [--circle-optional.radius float |
-                            --circle-optional.diameter float |
-                            --circle-optional.perimeter float]
-                            (--circle-required.radius float |
-                            --circle-required.diameter float |
-                            --circle-required.perimeter float)"""
-            else:
-                usage = """usage: example.py [-h] [--circle-optional.radius float |
-                         --circle-optional.diameter float |
-                         --circle-optional.perimeter float]
-                         (--circle-required.radius float |
-                         --circle-required.diameter float |
-                         --circle-required.perimeter float)"""
-        elif sys.version_info >= (3, 13):
-            usage = """usage: example.py [-h] [--circle-optional.radius float |
+        usage = (
+            """usage: example.py [-h] [--circle-optional.radius float |
                   --circle-optional.diameter float |
                   --circle-optional.perimeter float]
                   (--circle-required.radius float |
                   --circle-required.diameter float |
                   --circle-required.perimeter float)"""
-        else:
-            usage = """usage: example.py [-h]
+            if sys.version_info >= (3, 13)
+            else """usage: example.py [-h]
                   [--circle-optional.radius float | --circle-optional.diameter float | --circle-optional.perimeter float]
                   (--circle-required.radius float | --circle-required.diameter float | --circle-required.perimeter float)"""
+        )
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""{usage}
 
 {ARGPARSE_OPTIONS_TEXT}:
@@ -2604,7 +2525,7 @@ def test_cli_kebab_case(capsys, monkeypatch):
         sub_other_subcmd: CliSubCommand[DeepSubModel]
         sub_arg: str
 
-    class Root(BaseModel):
+    class Root(BaseModel, cli_prog_name='example.py'):
         root_subcmd: CliSubCommand[SubModel]
         other_subcmd: CliSubCommand[SubModel]
         root_arg: str
@@ -2648,7 +2569,7 @@ def test_cli_kebab_case(capsys, monkeypatch):
         with pytest.raises(SystemExit):
             CliApp.run(Root)
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
+            capsys.readouterr().out
             == f"""usage: example.py [-h] --root-arg str {{root-subcmd,other-subcmd}} ...
 
 {ARGPARSE_OPTIONS_TEXT}:
@@ -2662,22 +2583,13 @@ subcommands:
 """
         )
 
-        if PYTHON_3_14:
-            if IS_WINDOWS:
-                usage = """usage: example.py root-subcmd [-h] --sub-arg str
-                                        {sub-subcmd,sub-other-subcmd} ..."""
-            else:
-                usage = """usage: example.py root-subcmd [-h] --sub-arg str
-                                     {sub-subcmd,sub-other-subcmd} ..."""
-        else:
-            usage = """usage: example.py root-subcmd [-h] --sub-arg str
-                              {sub-subcmd,sub-other-subcmd} ..."""
         m.setattr(sys, 'argv', ['example.py', 'root-subcmd', '--help'])
         with pytest.raises(SystemExit):
             CliApp.run(Root)
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
-            == f"""{usage}
+            capsys.readouterr().out
+            == f"""usage: example.py root-subcmd [-h] --sub-arg str
+                              {{sub-subcmd,sub-other-subcmd}} ...
 
 {ARGPARSE_OPTIONS_TEXT}:
   -h, --help            show this help message and exit
@@ -2690,21 +2602,12 @@ subcommands:
 """
         )
 
-        if PYTHON_3_14:
-            if IS_WINDOWS:
-                usage = """usage: example.py root-subcmd sub-subcmd [-h] --deep-arg str
-                                                   DEEP-POS-ARG"""
-            else:
-                usage = """usage: example.py root-subcmd sub-subcmd [-h] --deep-arg str
-                                                DEEP-POS-ARG"""
-        else:
-            usage = """usage: example.py root-subcmd sub-subcmd [-h] --deep-arg str DEEP-POS-ARG"""
         m.setattr(sys, 'argv', ['example.py', 'root-subcmd', 'sub-subcmd', '--help'])
         with pytest.raises(SystemExit):
             CliApp.run(Root)
         assert (
-            sanitize_cli_output(capsys.readouterr().out)
-            == f"""{usage}
+            capsys.readouterr().out
+            == f"""usage: example.py root-subcmd sub-subcmd [-h] --deep-arg str DEEP-POS-ARG
 
 positional arguments:
   DEEP-POS-ARG

